src/cmd/helm.rs

```diff
@@ -561,6 +588,150 @@ impl Helm {
 
         Ok(())
     }
+
+    pub fn prepare_chart_backup<P>(
+        &self,
+        workspace_root_dir: P,
+        chart: &ChartInfo,
+        envs: &[(&str, &str)],
+        backup_resources: Vec<String>,
+    ) -> Result<Vec<(String, String)>, HelmError>
+    where
+        P: AsRef<Path>,
+    {
+        let mut backups: Vec<(String, String)> = vec![];
+        for backup_resource in backup_resources {
+            match kubectl_get_resource_yaml(
+                &self.kubernetes_config,
+                envs.to_vec(),
+                backup_resource.as_str(),
+                Some(chart.get_namespace_string().as_str()),
+            ) {
+                Ok(content) => {
+                    if !content.contains("No resources found") {
+                        backups.push((backup_resource, content));
+                    };
+                }
+                Err(e) => {
+                    error!("Kubectl error: {:?}", e.message_safe())
+                }
+            };
+        }
+
+        let mut backup_infos: Vec<(String, String)> = vec![];
+
+        if backups.len() == 0 {
+            return Ok(backup_infos);
+        }
+
+        for backup in backups.clone() {
+            match create_yaml_backup_file(
+                workspace_root_dir.as_ref(),
+                chart.name.to_string(),
+                Some(backup.0.clone()),
```

Comment:
OPT: Not very clear at first read. Maybe worth using a struct avoiding to have to index like this.